### PR TITLE
Full scale project refactoring (files, classes and folders)

### PR DESCRIPTION
**Pull request by Kjelli**
This is a big one. The most notable changes are:

### Class name and location restructuring
- _Renamed and moved forms:_
  - _fAbout -> Forms/AboutForm_
  - _fItemEdit -> Forms/ItemEditForm_
  - _fItemSelect -> Forms/ItemSelectorForm_
  - _fLogicEdit -> Forms/LogicEditorForm_
  - _fMain -> Forms/MainForm_
  - _fManual -> Forms/ManualForm_
  - _NewItemForm -> Forms/NewItemForm_

- _Renamed and moved most partial classes from `RomFuncs` to standalone utility classes, and matching them with their previous names:_
  - _AudioSeq -> Utils/AudioSequenceUtils_
  - _Base36 -> Utils/Base36Utils_
  - _Entrance -> Utils/EntranceUtils_
  - _ItemSwap -> Utils/ItemSwapUtils_
  - _Message -> Utils/MessageUtils_
  - _Obj -> Utils/ObjUtils_
  - _RW\_Helpers -> Utils/ReadWriteUtils_
  - _Resource -> Utils/ResourceUtils_
  - _ROM -> Utils/RomUtils_
  - _Scenes -> Utils/SceneUtils_
  - _Tunics -> Utils/TunicUtils_
  - _VC_Inject -> Utils/VCInjectionUtils_
  - ...
- _Extracted common models into separate classes:_
  - Rom models:
    - _Actor -> Models/Rom/Actor_
    - _BottleCatchEntry -> Models/Rom/BottleCatchEntry_
    - _CollisionTri -> Models/Rom/CollisionTri_
    - _Enemy -> Models/Rom/Enemy_
    - _GetItemEntry -> Models/Rom/GetItemEntry_
    - _Map -> Models/Rom/Map_
    - _MMFile -> Models/Rom/MMFile_
    - _MMMessage -> Models/Rom/MMMessage_
    - _MMSequence -> Models/Rom/MMSequence_
    - _Scene -> Models/Rom/Scene_
    - _SequenceInfo -> Models/Rom/SequenceInfo_
  - Settings models:
    - _Models/Character -> Models/Settings/Character_
    - _Models/Character -> Models/Settings/Character_
    - _Models/DamageEffect -> Models/Settings/DamageEffect_
    - _Models/DamageMode -> Models/Settings/DamageMode_
    - _Models/FloorType -> Models/Settings/FloorType_
    - _Models/LogicMode -> Models/Settings/LogicMode_
    - _Models/MovementMode -> Models/Settings/MovementMode_
    - _Models/Settings -> Models/Settings/Settings_
    - _Models/TatlColorSchema -> Models/Settings/TatlColorSchema_
  - Vector models:
    - _vec16 -> Models/Vectors/vec16_
    - _vecf32 -> Models/Vectors/vecf32_

- _Extracted common data that is read/write from/to everywhere into static class RomData (it's not a good permanent solution, but should be contained where relevant. At this point it is more tracable, and it only contains data.)_

- _ Moved resources (obj, addresses, mods, music) to subfolder Resources _

### Changes in the internal structure and flow of code
The Main Form has a privately referenced `Randomizer` (prev. named `Shuffle`), and settings from checkbox+dropdown selections. When user prompts to randomize, the `Randomizer` is responsible for doing all the randomization before its privately referenced `Builder` instance takes care of ROM writing (via `MakeRom(...)`).

### TL:DR
The project structure is **MUCH** cleaner with the changes made here (read: in my opinion), but is not perfect. I have put alot of effort into achieving better state of readability and structure. If something in this changeset is not preferable; instead of asking 'why is x y' and stating 'this is crap', feel free to provide constructive feedback.

I have tested the randomizer in several cases, and seems to be working as it did before - but having said that, I urge you to test it for yourself and let me know if I broke something.

### CODE REVIEW

File: Models/Spoiler.cs [hunk 1]
```diff
@@ -19,11 +19,9 @@ public class Spoiler
 
         public List<ItemObject> ItemList { get; set; }
 
-        public ReadOnlyCollection<string> ITEM_NAMES { get; set; }
+        public int[] NewEntranceIndices { get; set; }
```
Comment (mzxrules) [inline Models/Spoiler.cs]: should be NewDestinationIndices. entrance index refers specifically to the two byte index values used to spawn Link into the game world
